stackslib/src/net/api/getstxtransfercost.rs

```diff
@@ -92,9 +119,58 @@ impl RPCRequestHandler for RPCGetStxTransferCostRequestHandler {
         _contents: HttpRequestContents,
         node: &mut StacksNodeState,
     ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
-        // todo -- need to actually estimate the cost / length for token transfers
-        //   right now, it just uses the minimum.
-        let fee = MINIMUM_TX_FEE_RATE_PER_BYTE;
+        let estimated_len = self
+            .estimated_len
+            .take()
+            .ok_or(NetError::SendError("`estimated_len` not set".into()))?;
+
+        let fee_resp = node.with_node_state(|_network, sortdb, _chainstate, _mempool, rpc_args| {
+            let tip = self.get_canonical_burn_chain_tip(&preamble, sortdb)?;
+            let stacks_epoch = self.get_stacks_epoch(&preamble, sortdb, tip.block_height)?;
+
+            if let Some((_, fee_estimator, metric)) = rpc_args.get_estimators_ref() {
+                // STX transfer transactions have zero runtime cost
+                let estimated_cost = ExecutionCost::zero();
```

Comment:
I think they do have a non-zero cost. From costs-3:

```
(define-read-only (cost_stx_transfer (n uint))
    {
        runtime: u4640,
        write_length: u1,
        write_count: u1,
        read_count: u1,
        read_length: u1
    })
```

Reply:
That's for the Clarity function `(stx-transfer? ..)`.  The bare STX-transfer transaction has zero runtime cost, since it directly calls `stx_transfer_consolidated()` via an `Environment` (which only consumes VM memory, not runtime).

You can see for yourself as follows.  Consider the STX-transfer handling code:

```rust
let cost_before = clarity_tx.cost_so_far();
let (value, _asset_map, events) = clarity_tx
    .run_stx_transfer(
        &origin_account.principal,
        addr,
        u128::from(*amount),
        &BuffData {
            data: Vec::from(memo.0.clone()),
        },
    )
    .map_err(Error::ClarityError)?;

let mut total_cost = clarity_tx.cost_so_far();
total_cost
    .sub(&cost_before)
    .expect("BUG: total block cost decreased");

let receipt = StacksTransactionReceipt::from_stx_transfer(
    tx.clone(),
    events,
    value,
    total_cost,
);
Ok(receipt)
```
And here's a sample receipt pulled from mainnet (note the execution cost fields -- they're all 0):

```
    {
      "tx_id": "0x5d40b24095eb0282b2f25fcc8bae8c639a636ca85d9b97878e3123655bda5c83",
      "nonce": 3340,
      "fee_rate": "300",
      "sender_address": "SP16242W0FA64J0HSAB866XK1XPBWFBX4HNRPXSRX",
      "sponsored": false,
      "post_condition_mode": "deny",
      "post_conditions": [],
      "anchor_mode": "any",
      "is_unanchored": false,
      "block_hash": "0xf0d51285860650e6590b519977bf6087f9a6c98d75ceeffe4ac13c58f76f9dc9",
      "parent_block_hash": "0x4aabb7c0f31bc1ec13b42f61ddb5d430702843d2e69c96d652b7ff51d333848a",
      "block_height": 149171,
      "block_time": 1715102709,
      "block_time_iso": "2024-05-07T17:25:09.000Z",
      "burn_block_time": 1715102652,
      "burn_block_time_iso": "2024-05-07T17:24:12.000Z",
      "parent_burn_block_time": 1715102607,
      "parent_burn_block_time_iso": "2024-05-07T17:23:27.000Z",
      "canonical": true,
      "tx_index": 385,
      "tx_status": "success",
      "tx_result": {
        "hex": "0x0703",
        "repr": "(ok true)"
      },
      "microblock_hash": "0x",
      "microblock_sequence": 2147483647,
      "microblock_canonical": true,
      "event_count": 1,
      "events": [],
      "execution_cost_read_count": 0,
      "execution_cost_read_length": 0,
      "execution_cost_runtime": 0,
      "execution_cost_write_count": 0,
      "execution_cost_write_length": 0,
      "tx_type": "token_transfer",
      "token_transfer": {
        "recipient_address": "SP3WFJ0FT70HXVHS36C8WHFEVN6SRAT5P1B2MN4HC",
        "amount": "1",
        "memo": "0x6d627272723130300000000000000000000000000000000000000000000000000000"
      }
    },
```

Reply:
Oh interesting, thanks for that. I forgot that was treated differently. I guess we talked about this before when we were discussing how STX-20 transfers were unfairly cheap.